dialoptions.go

```diff
@@ -413,6 +413,9 @@ func WithTimeout(d time.Duration) DialOption {
 // connections. If FailOnNonTempDialError() is set to true, and an error is
 // returned by f, gRPC checks the error's Temporary() method to decide if it
 // should try to reconnect to the network address.
+//
+// Note: Go overrides the OS defaults for TCP keepalive interval to 15s.
+// To retain the OS defaults, pass in a custom dialer with KeepAlive set to a negative duration.
```

Comment:
We should elaborate be a bit more here. Maybe something like:

// To retain OS defaults, use a net.Dialer with the KeepAlive field set to a negative value.